docker/build-tools/Dockerfile

```diff
@@ -94,38 +92,44 @@ FROM ubuntu:xenial as base_os_context
 RUN apt-get update && apt-get install -y --no-install-recommends \
     ca-certificates \
     curl \
-	gcc \
+    gcc \
     git \
-	libc-dev \
-	make \
-	pkg-config \
-    wget \
-    && rm -rf /var/lib/apt/lists/*
+    libc-dev \
+    make \
+    pkg-config \
+    wget
 
-# Not in a sharing mood
+# Clean up stuff we don't need in the final image
+RUN rm -rf /var/lib/apt/lists/*
 RUN rm -fr /usr/share
 
 # Prepare final output image
 FROM scratch
 
 # Setup to run the Go compiler with host-side build cache, test cache, and module cache
 ENV GO111MODULE=on
+ENV GOPROXY=https://proxy.golang.org
 ENV GOPATH=/go
 ENV GOCACHE=/gocache
 ENV PATH /go/bin:$PATH
 
 COPY --from=base_os_context / /
-COPY --from=build_tools_context /out/ /
 COPY --from=golang_context /usr/local/go/ /go/
+COPY --from=build_tools_context /out/ /
 
-RUN mkdir -p /gocache
-RUN mkdir -p /go/pkg/mod
+RUN rm -fr /go/doc && \
```

Comment:
these should be deleted at the end of golang_context if they are not needed.  This will reduce the final image a bit - as these files as is will end up consuming disk space and storage transfer data.

Reply:
Done.